src/net.h

```diff
@@ -725,6 +725,7 @@ class CNodeStats
     // Name of the network the peer connected through
     std::string m_network;
     uint32_t m_mapped_as;
+    ConnectionType m_conn_type;
```

Comment:
a60f67b2fb0eede205db8cce11ad0e6f42dbc90e

This was suggested here https://github.com/bitcoin/bitcoin/pull/19725#discussion_r473361238 and now makes 
`m_conn_type_string` redundant, don't makes sense to keep both TBH. Since `m_conn_type_string` is only used in `getpeerinfo` then drop it and refactor `ConnectionTypeAsString` to take `conn_type` instead?

Reply:
Concept ACK, but RPC changes (refactoring) can only be made in the main repo

Reply:
Done, I think. My initial version of this PR did not touch any non-GUI code. Should I close this PR and re-open in the main repo? (I'm not sure it's worth losing the review above.)

Reply:
Maybe just open a pr in the main repo with the first and last commits?

Reply:
IMO this isn't a RPC change, it's just a member refactor in a net struct, and the refactor is motivated by this GUI change. Otherwise this can go forward in 2 ways I think, either leave this as it was, and then in main repo drop the redundant member, or 1st refactor and make this dependant of that PR.

Reply:
Yes, this doesn't change the RPC API.

Please let me know which option is best:

1. split it into two pulls, one here, gated by a another one in the other repo
2. move the pull to the other repo
3. revert to the original version that didn't touch main repo code
4. leave this as-is

Reply:
I'm not sure the refactoring changes make sense without the whole. Opened the same pull in https://github.com/bitcoin/bitcoin/pull/20778 and will let the maintainers decide.

Reply:
Just noting that redundancy is not a blocker, but could be avoided.